app/workers/search_for_external_registry_bikes_worker.rb

```diff
@@ -0,0 +1,28 @@
+class SearchForExternalRegistryBikesWorker < ScheduledWorker
+  prepend ScheduledWorkerRecorder
```

Comment:
This is clever and I support doing it this way.

However, please do this in a separate PR, include it in `ScheduledWorker` itself, and remove all the calls to `record_scheduler_started` and `_finished` in the existing scheduled workers

Reply:
> include it in ScheduledWorker itself

NB: That's not how `prepend` works — it has to be added to leaf / child classes to be inserted into the inheritance chain correctly.

```rb
UpdateExpiredInvoiceWorker.new.perform

#    6: def perform
#    7:   binding.pry
#    8:   # Invoice.should_expire.each { |invoice| invoice.update_attributes(updated_at: Time.current) # }

self.class.ancestors
# => [
#  [ 0] UpdateExpiredInvoiceWorker < ScheduledWorker,
#  [ 1] ScheduledWorkerRecorder,
#  [ 2] ScheduledWorker < ApplicationWorker,
#  [ 3] ApplicationWorker < Object,
```